consensus/enclave/impl/src/lib.rs

```diff
@@ -799,13 +872,16 @@ impl ConsensusEnclave for SgxConsensusEnclave {
 /// * `parent_block` - The parent block.
 /// * `transactions` - The transactions that are included in the current block.
 /// * `amount` - Output amount.
+/// * `counter` - An additional counter used to disambiguate hashes, when the
+///   same amount is minted repeatedly
 fn mint_output<T: Digestible>(
     block_version: BlockVersion,
     recipient: &PublicAddress,
     domain_tag: &'static [u8],
     parent_block: &Block,
     transactions: &[T],
     amount: Amount,
+    counter: usize,
```

Comment:
@eranrund i had to add this, or else in some corner cases the network can get stuck:

* If the accumulated fee is like 3 * u64::MAX, we would mint 3 outputs of the same value, u64::MAX. But all these things we are hashing would be the same, and so they would have the same tx private key, and the tx public key would collide, causing network to get stuck.
* After the other change in this PR, this causes it to get stuck in slam, because it mints extra fee outputs.
* If someone submits two mint tx operations, both for the same amount and with the same recipient, those minted outputs would also have the same tx private key and the same tx public key. The counter fixes this.

Reply:
Good catch, this all makes sense to me.

Reply:
Makes sense. Please document this rationale.